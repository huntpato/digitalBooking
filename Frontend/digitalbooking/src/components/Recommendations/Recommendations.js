import React, {useState} from "react";
import { ReCard } from "../ReCard/ReCard";
import useFetch from "use-http";
import styles from "./Recommendations.module.css";

export const Recommendations = () => {
  
  const options = { headers:{
    "X-Requested-With": "XMLHttpRequest"
  } };
  const url = "https://cors-everywhere-me.herokuapp.com/g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/products";
  const { loading, data = [] } = useFetch(url, options, []);
  const { recommendation, recommendationCards, recommendationtitle, recommendationContainer, loadMore } = styles;
  const [products, setProducts] = useState(4);

  const handleClick = () => {
    
    setProducts(prevProducts => prevProducts + 4)
    
  }

  return (
    <div>
      <section className={recommendation}>
        <div className={recommendationContainer}>
          <h4 className={recommendationtitle}>Recomendaciones</h4>
          <div className={recommendationCards}>
            {loading && <p>Cargando Recomendaciones...</p>}
            {data.slice(0,products).map((product, index) => (
              <ReCard
                key={index}
                type={product.productCategory.name}
                title={product.productName}
                img={product.photoGallery[0]}
                alt={product.productDescription}
                text={product.productDescription}
                loc={product.location}
                rate={product.review}
                id={product.id}
                services={product.services}
                latlong={product.coordinates}
              />
            ))}
          </div>
          {data[products] && <button className={loadMore} onClick={handleClick}>Ver m√°s</button>}    
        </div>
      </section>
    </div>
  );
};
