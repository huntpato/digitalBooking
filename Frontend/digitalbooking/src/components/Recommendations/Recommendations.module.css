.recommendation { 
  background-color: var(--background-color2);
  display: flex;
  flex-direction: row;
  justify-content: center; 
}

.recommendationContainer{
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  margin: 30px 0px 30px 0px;
  width: 90vw;
}

.recommendationtitle {
  
  margin-bottom: 10px;
  
}

.recommendationCards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;

}

.loadMore{
  align-self: center;
  background-color: var(--principal-color);
  border: 1.4px solid var(--principal-color);
  border-radius: 5px;
  color: var(--background-white);
  cursor: pointer;
  display: flex;
  font-weight: 600;
  justify-content: center;
  margin: 30px 0px 0 0px;
  padding: 7px 0px 6px 0px;
  text-align: center;
  transition: 0.5s;
  width:30%
}

.loadMore:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.97);
}

.loadMore:hover {
  background-color: var(--background-white);
  color: var(--principal-color);
} 

.loadMore:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.98);
}

@media (max-width: 768px) {

  .recommendation {
    align-content: center;
    background-color: var(--background-color);
    display: flex;
    flex-direction: column;
    flex-wrap: wrap;
    width: 100%;
  }

  .recommendation h4 {
    margin-bottom: 10px;
    margin-left: 0px;
  }

  .recommendationCards {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  .loadMore{
    width: 50%;
  }
  
  
}

@media (max-width: 480px) {

  .loadMore{
    width: 96%;
  }
  
}
