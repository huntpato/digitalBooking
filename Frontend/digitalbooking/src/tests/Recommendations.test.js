import React from "react";
import { mount } from "enzyme";
import { Recommendations } from "../components/Recommendations/Recommendations";
import "@testing-library/jest-dom";
import { expect } from "@jest/globals";

global.fetch = jest.fn(()=> Promise.resolve({
  json: ()=>Promise.resolve({
      "id": 1,
      "productName": "Amazing Luxurious Apt-Palermo Soho",
      "productDescription": "Luxury building in the heart of the trendy Palermo Soho. 24-hour security",
      "publicationDate": "2009-04-13",
      "photoGallery": [
          "https://a0.muscache.com/im/pictures/c5cd9326-df4d-4b63-960b-4d64a95d04a1.jpg",
          "https://a0.muscache.com/im/pictures/dcea15c9-1165-4012-ae05-291744953799.jpg",
          "https://a0.muscache.com/im/pictures/ad19a0b4-ab77-4e5b-ac5a-eb64d03c7118.jpg",
          "https://a0.muscache.com/im/pictures/3df8765e-5013-45f4-9eb8-2d2d2eb20e82.jpg",
          "https://a0.muscache.com/im/pictures/09f10f8f-5215-455a-bb6f-3df43be49e6d.jpg"
      ],
      "user": {
          "id": 1,
          "nombre": "admin",
          "apellido": "admin",
          "email": "admin@dh.com"
      },
      "city": {
          "id": 1,
          "name": "Buenos Aires, Capital Federal",
          "country": "Argentina"
      },
      "coordinates": [
          "-34.58184",
          "-58.42415"
      ],
      "productCategory": {
          "id": 4,
          "name": "Islas",
          "photoUrl": "https://media.istockphoto.com/photos/young-woman-walks-towards-beautiful-beach-picture-id1311063862?b=1&k=20&m=1311063862&s=170667a&w=0&h=ai-8qth6hlECiFoQR4g9Wt1cZHdrvsCsSAnyzGP2d3E=",
          "description": "Islas",
          "countProducts": "http://g4bookingapp-env-2.eba-4mbmxg4f.us-east-1.elasticbeanstalk.com/category/count/Islas"
      },
      "maxGuess": 2,
      "rooms": 2,
      "services": [
          "Wifi",
          "Cable TV",
          "kitchen",
          "parking",
          "jacuzzi"
      ],
      "minNights": 3,
      "maxNights": 30,
      "score": 4.7,
      "review": 4,
      "pricePerNight": 6408,
      "productAvailable": true
  })
}))

jest.mock('react-router-dom', () => ({
  useParams: jest.fn().mockReturnValue({ nifUuid: 'nif123' }),
  useHistory: jest.fn(),
  useContext: jest.fn().mockReturnValue({visibility: true}),
  useLocation:jest.fn().mockReturnValue({
    pathname: "/register",
    search: "",
    hash: "",
    state: null,
    key: "5nvxpbdafa",
  }),
}));

describe("Pruebas en componente  Recommendations />", () => {
  const wrapper = mount(<Recommendations />);

  test("Comprobar que  Recommendations /> renderiza correctamente.", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Verificar que el componente carga la clase de CSS correctamente", () => {
    const section = wrapper.find("section");
    expect(section.hasClass("recommendation")).toBe(true);
  });

  
  test("Verificar que muestra el mensaje de cargando recomendaciones", () => {
    const section = wrapper.find("p").text();
    expect(section).toContain("Cargando Recomendaciones...");
  });

});
