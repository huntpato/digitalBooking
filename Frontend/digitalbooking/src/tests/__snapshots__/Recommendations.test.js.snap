// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente  Recommendations /> Comprobar que  Recommendations /> renderiza correctamente. 1`] = `
<div>
  <section
    className="recommendation"
  >
    <div
      className="recommendationContainer"
    >
      <h4
        className="recommendationtitle"
      >
        Recomendaciones
      </h4>
      <div
        className="recommendationCards"
      >
        <p>
          Cargando Recomendaciones...
        </p>
      </div>
    </div>
  </section>
</div>
`;
